libs/@blockprotocol/type-system/rust/src/schema/data_type/constraint/mod.rs

```diff
@@ -6,32 +7,216 @@ mod number;
 mod object;
 mod string;
 
-use error_stack::{Report, ResultExt, bail};
+use error_stack::{Report, bail};
 use serde::{Deserialize, Serialize};
 use serde_json::Value as JsonValue;
 
-pub(crate) use self::{
-    array::ArraySchema, boolean::BooleanSchema, error::ConstraintError, null::NullSchema,
-    number::NumberSchema, object::ObjectSchema, string::StringSchema,
+pub use self::{
+    any_of::{AnyOfConstraints, AnyOfSchema},
+    array::{ArrayConstraints, ArraySchema, ArrayTypeTag, ArrayValidationError, TupleConstraints},
+    boolean::BooleanTypeTag,
+    error::ConstraintError,
+    null::NullTypeTag,
+    number::{NumberConstraints, NumberSchema, NumberTypeTag, NumberValidationError},
+    object::ObjectTypeTag,
+    string::{
+        StringConstraints, StringFormat, StringFormatError, StringSchema, StringTypeTag,
+        StringValidationError,
+    },
 };
-use crate::schema::JsonSchemaValueType;
+use crate::schema::{JsonSchemaValueType, ValueLabel};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SimpleTypedValueSchema {
```

Comment:
Can we discuss these names? Not sure about them:
- why `SimpleValueSchema` to cover both the `anyOf` and the other stuff? Can't this just be called `ValueSchema`?
- re. `SimpleTypedValueSchema`... isn't all this stuff `Typed`? Maybe this could be `SingleValueSchema` instead, to distinguish between `anyOf`, which is multiple schemas? And then `SingleValueConstraint`